libp2p/stream/bufferstream.nim

```diff
@@ -107,10 +109,14 @@ method readOnce*(s: BufferStream,
 
   if rbytes < nbytes:
     # There's space in the buffer - consume some data from the read queue
-    trace "popping readQueue", s, rbytes, nbytes
-    let buf = await s.readQueue.popFirst()
-
-    if buf.len == 0 or s.isEof: # Another task might have set EOF!
+    s.reading = true
+    let buf =
+      try:
+        await s.readQueue.popFirst()
```

Comment:
try/finally potential issues here?

Reply:
as long as it's not nested and not in a loop, it's fine